### PR TITLE
adding checks to validating auth-provider

### LINKED ISSUE #80: Validate AuthProvider Helm Values
For OpenMetadata, we need to restrict Auth Provider Values that we provide under `global.authentication.provider` with only below valid options -
- basic
- auth0
- azure
- custom-oidc
- google
- okta
- aws-cognito

Validate the input and error out if the value does not meet the above options with a meaningful error.
Add the validation logic with go templates.

### PR DESCRIPTION
**Pull request by Ashok810**
### Describe your changes :

fixes #80 

#
### Type of change :

- [x] Improvement
- [x] New feature
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developer) document.
- [ ] I have performed a self-review of my own. 
- [ ] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: charts/openmetadata/templates/deployment.yaml [hunk 1]
```diff
@@ -71,6 +71,7 @@ spec:
             value: "{{ .Values.global.openmetadata.port }}"
           - name: SERVER_ADMIN_PORT
             value: "{{ .Values.global.openmetadata.adminPort }}"
+            {{- include "check_provider_value" . }}
```
Comment (akash-jain-10) [inline charts/openmetadata/templates/deployment.yaml:74]: This is not required!

Comment (akash-jain-10) [reply charts/openmetadata/templates/deployment.yaml:74]: Let's remove this code!

File: charts/openmetadata/templates/_validate-values.tpl [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{{- define "check_provider_value" }}
```
Comment (akash-jain-10) [inline charts/openmetadata/templates/_validate-values.tpl:1]: Can you please update the filename to `validate-values.tpl` !

It seems partial files are do not support invocation of functions defined for Helm!

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: charts/openmetadata/templates/deployment.yaml (hunk 1 above)
Comment (akash-jain-10) [reply charts/openmetadata/templates/deployment.yaml:74]: Let's remove `{{- include "check_provider_value" . }}` this piece of code!

**Review by akash-jain-10 — CHANGES_REQUESTED** [review]

File: charts/openmetadata/templates/deployment.yaml (hunk 1 above)
Comment (Ashok810) [reply charts/openmetadata/templates/deployment.yaml]: removed the code {{- include "check_provider_value" . }}

File: charts/openmetadata/templates/_validate-values.tpl (hunk 2 above)
Comment (Ashok810) [reply charts/openmetadata/templates/_validate-values.tpl]: updated the filename to validate-values.tpl

**Review by akash-jain-10 — APPROVED** [review]

**Pull request merged**